### DISCUSSION TITLE
Shelly Plus 2PM Tasmota Fail

### DISCUSSION THREAD

**Discussion by PikAss2017** [arendst/Tasmota#21831] [Q&A]
Hey Everyone, i hope i can explain my Problem in easy Words. This is my first Thread in English :-)

I tried to upgrade Tasmota 12.5.0 to the latest version on my Shelly Plus 2PM Device, but he told me, that my flash size is not enough. I found some entries about that problem and tried a bit with this, BUT
I dont know why, but i found the Manage File System Button and deleted some Files ... i think this is my problem now.

![image](https://github.com/user-attachments/assets/2f71e5ed-9bc6-4d8c-862b-1d45f885e419)

After that i tried to hit upgrade 3 in the console -> Errormessage, after that upgrade 2 -> Device Dead

With the Power Cycle Method, my Device came back to life. I checked more Threads about my Problem and found the Partition Wizard Safeboot Magic Method, i tried to do this on my Device BUT:

![image](https://github.com/user-attachments/assets/8985ac84-0465-486d-9629-e1736edac4a4)

Checking the Partition Manger:

![image](https://github.com/user-attachments/assets/1ad3f45e-9431-4f32-934f-26a5a17d3d78)

Checking Console after Power Cycle Reset again:

![image](https://github.com/user-attachments/assets/9cd4b6a8-9f62-4eca-a158-9402cb317f73)

I think i deleted some partition info files with the Manage File System Button and now he cant do anything.
Is there any way to get these informations back to the system via Web / OTA?

I ordered an CH340G Adapter via Amazon to Flash my Device again over the PC USB Connection, maybe it is the only way to get that device back to life.

**Comment by Jason2866** [comment]
You can try if PartitionManager fixes the somehow broken partition table. Load https://github.com/arendst/Tasmota/blob/development/tasmota/berry/modules/Partition_Manager.tapp
and try to resize a partition. 16kb change is enough. If this is successful. Try PartitionWizard again to convert to safeboot scheme.

**Reply by PikAss2017** [reply]
Thank you for your comment.  You can see how it looks like when i open the Partition Manager.
There is only the button Empty Partition, nothing else. This Button has no Action.

**Reply by Jason2866** [reply]
So wait for the USB serial adapter to flash by wire.

**Reply by sfromis** [reply]
FTR: I ran a test where I flashed Tasmota version 12.5 on a device, and then attempted to upgrade to safeboot partitioning scheme. It did indeed not work when using current Partition_Wizard.tapp, but using the old version from 12.5 too, I made it work when I also changed the two URL references at the bottom of the page to refer to release-12.5.0 instead of just release. After that, it worked, no issues completing the upgrade.

(Not knowing the state of your setup, IDK if you can do the above)
